lib/ical/recur_iterator.js

```diff
@@ -974,6 +972,9 @@ ICAL.RecurIterator = (function() {
             }
           }
         }
+        // Return dates in order of occurrence (1,2,3, etc...)
+        // instead of by weekday (su, mo, etc..).
+        this.days.sort(function(a,b){return a - b;});
```

Comment:
Ah of course! I am wrong. I was testing with numbers 1,2,3 and obviously that didn't expose any issues!